src/customHooks/useVideoDetail.js

```diff
@@ -0,0 +1,30 @@
+import { useContext, useState, useEffect, useMemo } from 'react'
+import { useParams } from 'react-router'
+import { YoutubeContext } from '../context/YoutubeContext'
+import { getArrayFromLocalStorage } from '../utils/localStorage'
+
+export default function useVideoDetail() {
+	const { videoId } = useParams()
+	const { videos, setVideosContext } = useContext(YoutubeContext)
+	const [currentVideo, setCurrentVideo] = useState({ snippet: { title: '...' } })
+
+	useEffect(() => {
+		let videosTmp
+		if (videos.length) videosTmp = [...videos]
+		else {
+			videosTmp = getArrayFromLocalStorage('videos') || []
+			if (videosTmp.length) setVideosContext(videosTmp)
+		}
+
+		const videoFound = videosTmp.find(
+			video => String(video.id.videoId) === String(videoId)
+		)
+		if (videoFound) setCurrentVideo(videoFound)
+	}, [videoId, videos, setVideosContext]) // It'll be executed once
+
+	const relatedVideos = useMemo(() => {
```

Comment:
Just as a side note, I think there's an specific endpoint to gather the related videos of a specific videos, maybe that could help you out in order to visualize real results and you can trigger the fetching of the related videos everytime the url changes